containers/proxy-httpd-image/prepare_target.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# Removes any unnecessary files and packages before moving to the next build stage
+
+set -xe
+
+# remove rpm-build and its dependencies
+rpm -e rpm-build --nodeps
+rpm -e dwz make gcc patch diffutils python-rpm-macros systemd-rpm-macros glibc-locale
```

Comment:
When are such packages getting installed? During the zypper call at https://github.com/uyuni-project/uyuni/pull/5363/files#diff-c81a9cc5087e00f7ef306e86a8aaaca03bcfbf348a9f5d1b0aefe11dd58fcc6fR14?

If so, you could consider calling this script at that same call `with && sh prepare_target.sh`

Reply:
The `spacewalk-certs-tools` is responsible for installing it. Again I believe it's a necessary dependency of the package, or is it @cbosdo ?

regarding merging the calls, me personally, I would rather keep them separate since we use multistage build anyway and merging calls wouldn't bring us any improvement in size. At least for me it's more readable, but if we have some other conventions for such things, I'm totally open for it :smile:

Reply:
> regarding merging the calls, me personally, I would rather keep them separate since we use multistage build anyway and merging calls wouldn't bring us any improvement in size. At least for me it's more readable, but if we have some other conventions for such things, I'm totally open for it smile

Well, the benefit is that you won't ever had such packages on any of the stages, if you don't need them for the final image.

If right now there are no size benefits that's probably because a later stage is bigger than the stage before prepare_target.sh is called.

Reply:
You're totally right. 
If I understand it correctly, at least starting from docker >17.05 we got two mutually excluding options. 
1. we can either merge together the _add_ and _remove_ operations so that we don't create additional layers/commits in between and thus saving us space. 
2. or as an alternative we can use multistage builds where we copy over only the files we need from the head of a previous stage, not the layers/commits as such. 
Both options lead to the same image size, I've just tested :smiley:

Reply:
But surely I don't insist on any particular approach. I just decided to go for the best practices as they're declared out there.

Reply:
Ok I see the real benefit. We don't overuse the cache while building the image if we merge together the calls. So I'll merge it.